### PR TITLE
DFBW-3 Profile Component

### LINKED ISSUE #3: DFBW - Profile Component
# Profile Component

### Must have

Profile form with:

- Name input *
- Last name input *
- Email input *
- Profile photo
- Save button

Update password form with:

- "Cambiar contraseña" label
- Current password input *
- New Password input *
- Repeat new Password input *
- Update password button

\* Must be required

### Data Entry

- Profile Form data

### Process

- User can update his data and use the Save button to update it.

### Output

- User information updated in database
- Successful message

### Error Output

- Message with error

#### Frontend

- Check if form data is correct and complete, if not show error, otherwise send profile data and session data to user endpoint.

- Retrieve data from endpoint.
- In case of sucess show sucessful message and show new profile photo if applies, otherwise show error.

#### Backend

- Update user information in database, return success message or error message in case of error.

### Data Entry

- Update password form data

### Process

- User can update his password and use the "update password" button to update it.

### Output

- User password updated in database
- Successful message

### Error Output

- Generic message with error

#### Frontend

- Check if current password input match with the one saved in session data.
- Check if both new password and repeat new password match and have at least 8 characters.
- Send new password and auth data to auth endpoint.

- Retrieve message from backend and update the session with the new token.

#### Backend

- Retrieve info, check if user has permissions to update password.

- Try to update password, if user has permissions, send sucessful message and send a new token, otherwise send error.

### PR DESCRIPTION
**Pull request by omarefg**
* [x] I have followed the [Contributor guideline](https://github.com/Master-delta-force/bikeapp).
* [ ] Tests are included
* [ ] Documentation is changed or added
* [x] Related issue has been created
* [x] Commits messages follows [commit guideline](https://github.com/Master-delta-force/bikeapp) 

resolves #3

### CODE REVIEW

File: src/routes/profile.svelte [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+function toggleIsEditing () {
+  isEditing = !isEditing
+}
+
+</script>
+
+<style>
+  .profile_-container {
```
Comment (AryRosvall) [inline src/routes/profile.svelte:49]: I think here is a typo

File: src/components/atoms/Input.svelte [hunk 2]
```diff
@@ -4,11 +4,13 @@
   export let name;
```
Comment (AryRosvall) [inline src/components/atoms/Input.svelte:4]: Why this is not initialized and the other ones do?

File: src/components/atoms/Image.svelte [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <script>
     export let src
```
Comment (AryRosvall) [inline src/components/atoms/Image.svelte:2]: Why only one is initialized?

**Review by AryRosvall — CHANGES_REQUESTED** [review]

File: src/routes/profile.svelte (hunk 1 above)
Comment (omarefg) [reply src/routes/profile.svelte]: You're right, I already uploaded a fix for it.

File: src/components/atoms/Input.svelte (hunk 2 above)
Comment (omarefg) [reply src/components/atoms/Input.svelte:4]: Actually that component was not supposed to be on this PR. But I found needs to apply some changes to it for the profile page to use it, that's why it appears here. And I was trying to keep it as it was the most I could.

File: src/components/atoms/Image.svelte (hunk 3 above)
Comment (omarefg) [reply src/components/atoms/Image.svelte:2]: Actually, same answer that above haha. But as this one was written for me I can say that src prop should not be initialized by default. It is a mandatory prop for the image component.

**Review by AryRosvall — APPROVED** [review]

**Review by IntiDev — APPROVED** [review]

**Review by zikeron — APPROVED** [review]

**Pull request merged**